service/rest.go

```diff
@@ -84,7 +84,7 @@ func GetRESTv1App(evgService restAPIService) *gimlet.APIApp {
 	app.AddRoute("/tasks/{task_id}").Version(1).Get().Handler(rest.getTaskInfo).Wrap(middleware)
 	app.AddRoute("/tasks/{task_id}/status").Version(1).Get().Handler(rest.getTaskStatus).Wrap(middleware)
 	app.AddRoute("/versions/{version_id}").Version(1).Get().Handler(rest.getVersionInfo).Wrap(middleware)
-	app.AddRoute("/versions/{version_id}").Version(1).Patch().Handler(requireUser(rest.modifyVersionInfo, nil)).Wrap(middleware)
+	app.AddRoute("/versions/{version_id}").Version(1).Patch().Handler(requireUser(false, rest.modifyVersionInfo, nil)).Wrap(middleware)
```

Comment:
A little confused; do all the other routes already have auth as part of their middleware or handlers?  From what I remember for example the waterfall is supposed to be unauthed.

Reply:
To be honest I'm not sure of the context why this route in particular had this handler function wrapper on it... I just refactored the function when I was making changes to the way `requireUser` was used for the UI routes.

On a related note, the middleware currently used in this file is not doing what I initially thought and is only checking for a logged in user when the project context's project ref is private, or if the project context's patch is defined.

I've imported `NewRequireAuthHandler` to this file and applied it to all routes so now they should all actually be only available to logged in users.